syntax = "proto2";

package hit_consistency;

option cc_generic_services = true;
// The Raft service.
service RaftService {
    // Sends a greeting
    rpc prevote (RequestVoteRequest) returns (RequestVoteResponse) {}
}
  
message RequestVoteRequest {
    optional string server_id = 1;
    optional string peer_id = 2;
    optional int64 term = 3;
    optional int64 last_log_term = 4;
    optional int64 last_log_index = 5;
}

message RequestVoteResponse {
    optional int64 term = 1;
    optional bool granted = 2;
};